src/Routes/ListRoutesCommand.php

```diff
@@ -0,0 +1,231 @@
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_FORMAT,
+            'table'
+        );
+
+        $this->addOption(
+            'sort',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_SORT,
+            'name'
+        );
+
+        // Routing table filter options
+        $this->addOption(
+            'has-middleware',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_MIDDLEWARE,
+            false
+        );
+        $this->addOption(
+            'has-name',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_NAME,
+            false
+        );
+        $this->addOption(
+            'has-path',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_PATH,
+            false
+        );
+        $this->addOption(
+            'supports-method',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_SUPPORTS_METHOD,
+            false
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $result = 0;
+        if (empty($this->routes)) {
+            $output->writeln("There are no routes in the application's routing table.");
+            return $result;
+        }
+
+        $format = strtolower((string) $input->getOption('format'));
+
+        $sorter = $this->getSortOrder($input) === 'name'
+            ? new RouteSorterByName()
+            : new RouteSorterByPath();
+        usort($this->routes, $sorter);
+
+        $this->filterOptions = [
+            'method'     => strtolower((string) $input->getOption('supports-method')),
+            'middleware' => strtolower((string) $input->getOption('has-middleware')),
+            'name'       => strtolower((string) $input->getOption('has-name')),
+            'path'       => strtolower((string) $input->getOption('has-path')),
+        ];
+
+        switch ($format) {
+            case 'json':
+                $output->writeln(json_encode($this->getRows(true)));
+                $output->writeln(
+                    "Listing the application's routing table in JSON format."
+                );
+                break;
+            case 'table':
+            case '':
+                $table = new Table($output);
+                $table->setHeaderTitle('Routes')
+                    ->setHeaders(['Name', 'Path', 'Methods', 'Middleware'])
+                    ->setRows($this->getRows(false));
+                $table->render();
+                $output->writeln(
+                    "Listing the application's routing table in table format."
+                );
+                break;
+            default:
+                $result = -1;
+                $output->writeln(
+                    "Invalid output format supplied. Valid options are 'table' and 'json'"
+                );
```

Comment:
Perhaps just make it the default, or assert on `$format` before?

Reply:
Yah, I would prefer having `table` as a default. Thats basically how most of the composer CLI commands work as well 👍🏼 
Even options can contain defaults in `symfony/console` AFAIR. No need to have that empty string check.